docs: trace tests to requirements P+3

lskare opened the pull request:
add a section on how the implemented tests relate and trace back to the original requirements of the patch. The section shows that the 3 implemented tests fulfill all 5 initial requirements

Resolves #18

Commit:
docs: trace tests to requirements P+3

add a section on how the implemented tests relate and trace back to the original requirements of the patch. The section shows that the 3 implemented tests fulfill all 5 initial requirements

Resolves #18

Stenmarken commented on Fundsoft/report.md:
```diff
@@ -100,6 +100,20 @@ Scope (functionality and code affected).
 Optional (point 3): trace tests to requirements.
 
+* test_update_transitive()
+    * This test adds the dependency `requests` and later updates the sub-dependency `chardet`. Asserts that the sub-dependency is still not in the lock file and that the sub-dependency has been updated to the newest version from the virtual repository. Also asserts that the dependency `requests` has not been updated as this was not requested. 
```
The test checks that the dependency is not in the pyproject.toml file. All sub-dependencies will still be in the lock file.

Stenmarken commented on Fundsoft/report.md:
```diff
@@ -100,6 +100,20 @@ Scope (functionality and code affected).
+    * Fulfills requirements **ID 2**
+
+* test_update_transitive_non_transitive_dependencies()
+    * This test adds two dependencies `requests` and `pytz` and later updates both of these along with the sub-dependency `chardet`. It asserts that the sub-dependency is still not in the lock file and asserts that the sub-dependency and the two dependencies are both updated to newer versions from the virtual repository even when utilizing the flag "--allow_transitive". 
```
Again, the sub-dependency should not be in the pyproject.toml file. We expect it to be in the lock file regardless of whether we update it or not.

Stenmarken reviewed (commented):
Good job! I just had two comments, otherwise I thought everything was good!

Commit:
docs: change lock file to pyproject.toml

correct a minor but important typo by changing that "a sub-depencency is still not in the lock file" to "a sub-dependency is still not in the pyproject.toml file".

lskare commented:
Thanks for the feedback! I've now corrected this typo and written pyproject.toml file instead!

Stenmarken reviewed (approved):
Good job!
